### PR TITLE
linux: riscv32: Add support for the 64-bit time_t RV32 glibc port

### PR DESCRIPTION
**Pull request by alistair23**

### CODE REVIEW

**Comment by JohnTitor** [comment]
I'd rubber-stamp this as it seems like copy-pasted. And could you add the `riscv32gc-unknown-linux-gnu` there? I think it's worth to run CI for that target as well.

https://github.com/rust-lang/libc/blob/9c17cad7f0b03e3515f0cd5c56b5592b384cf9d4/ci/build.sh#L225-L227

**Comment by alistair23** [comment]
Done!

File: src/unix/linux_like/linux/gnu/b32/mod.rs [hunk 1]
```diff
@@ -278,6 +278,9 @@ cfg_if! {
     } else if #[cfg(target_arch = "sparc")] {
         mod sparc;
         pub use self::sparc::*;
+    } else if #[cfg(any(target_arch = "riscv32"))] {
```
Comment (JohnTitor) [inline src/unix/linux_like/linux/gnu/b32/mod.rs:281]:
```suggestion
    } else if #[cfg(target_arch = "riscv32")] {
```

**Comment by JohnTitor** [comment]
Oh, CI found an issue, seems some items are re-defined in the 32-bit space and CI complains about it. We should remove/tweak them following the actual implementation.

**Comment by JohnTitor** [comment]
Ugh, Pipelines is lazy, re-run it...

**Comment by alistair23** [comment]
Thanks, sorry I can't get the CI to run locally. I'll ping you here when I have the tests passing.

**Comment by alistair23** [comment]
Argh! Now I'm getting errors like this:

```rust
error: type alias is never used: `ino_t`
  --> src/unix/linux_like/linux/gnu/b32/riscv32/mod.rs:18:1
   |
18 | pub type ino_t = u64;
   | ^^^^^^^^^^^^^^^^^^^^^
```

I need to overwrite the `ino_t` type though. Any ideas?

**Comment by JohnTitor** [comment]
The easiest way is to use `cfg_if` macro, like:
```rust
cfg_if! {
    if #[cfg(target_arch = "riscv32")] {
        pub type ino_t = u64;
    } else {
        // ...
    }
}
```
at gnu/b32/mod.rs.

**Comment by alistair23** [comment]
Woo! CI is passing.

**Review by JohnTitor — APPROVED** [review]
Awesome, thanks!

**Pull request merged**